packages/template-typescript-minimal/app/components/_app-config/index.js

```diff
@@ -16,14 +16,25 @@ const options = {
     queryClientConfig: {
         defaultOptions: {
             queries: {
-                retry: false,
+                refetchOnWindowFocus: false,
                 staleTime: 2 * 1000,
-                ...(isServerSide ? {retryOnMount: false} : {})
-            },
-            mutations: {
-                retry: false
             }
         }
+    },
+    beforeHydrate: (data) => {
+        const now = Date.now()
+
+        // Helper to reset the data timestamp to time of app load.
+        const updateQueryTimeStamp = ({state}) => {
+            state.dataUpdatedAt = now
+        }
+
+        // Update serialized mutations and queries to ensure that the cached data is
+        // considered fresh on first load.
+        data?.mutations?.forEach(updateQueryTimeStamp)
+        data?.queries?.forEach(updateQueryTimeStamp)
+
+        return data
```

Comment:
How do you feel about creating a util in `commerce-sdk-react` that will do this for you. Maybe call it something like `resetQueryTimeStamp()` or `updateQueryTimeStamp(date)`. This code seems to be repeated a lot and might be a pain to change if required.

Reply:
thanks! I ended up calling it `resetDehydratedQueryTimeStamp`, because the `data` object is type `DehydratedState` from react-query.